OrbitQt/topdownwidget.cpp

```diff
@@ -65,6 +69,32 @@ static void CollapseChildrenRecursively(QTreeView* tree_view, const QModelIndex&
   }
 }
 
+static std::vector<std::shared_ptr<Module>> GetModulesFromIndices(
+    OrbitApp* app, const std::vector<QModelIndex>& indices) {
+  std::shared_ptr<Process> process = app->GetCaptureData().process();
+  CHECK(process != nullptr);
+
+  std::set<std::string> unique_module_paths;
+  for (const auto& index : indices) {
+    std::string module_path =
+        index.model()
+            ->index(index.row(), TopDownViewItemModel::kModule, index.parent())
+            .data(Qt::ToolTipRole)
+            .toString()
+            .toStdString();
+    unique_module_paths.insert(module_path);
+  }
```

Comment:
Have you considered storing the `std::shared_ptr<Module>` directly inside of `TopDownFunction` and adding a custom DataRole to the view such that you can directly access the modules from QModelIndexes?

I'm basically asking because adding a dependency to OrbitApp in TopDownWidget feels a bit wrong. Ideally the widget should get all its data from the model (TopDownView)

Reply:
Also using the ToolTipRole here seems to be brittle. Whenever someone wants to adjust the tooltip in the future this logic might be broken.

Reply:
> Have you considered storing the std::shared_ptr<Module> directly inside of TopDownFunction and adding a custom DataRole to the view such that you can directly access the modules from QModelIndexes?
I'm basically asking because adding a dependency to OrbitApp in TopDownWidget feels a bit wrong. Ideally the widget should get all its data from the model (TopDownView)

Personally I would prefer the view to have only a key to the data, not to store a pointer to the data itself, but maybe it's just me. But to answer them main point of your concern, I need `TopDownWidget` to depend on `OrbitApp` anyway, because that's the bridge to being able to perform actions that affect the rest of Orbit (in this case, loading modules).

Reply:
Ah I see.

Reply:
If you have ideas regarding (not) depending on `OrbitApp` let's discuss them, I don't exactly like such dependency either.

Reply:
In the meantime I added a custom data role to retrieve the module's path, please have a look.

Reply:
That looks good to me. Thanks. I don't realy know what's the best way to solve this.